compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -344,17 +346,32 @@ pub fn collect_crate_mono_items(
         let inlining_map: MTRef<'_, _> = &mut inlining_map;
 
         tcx.sess.time("monomorphization_collector_graph_walk", || {
-            par_for_each_in(roots, |root| {
-                let mut recursion_depths = DefIdMap::default();
-                collect_items_rec(
-                    tcx,
-                    dummy_spanned(root),
-                    visited,
-                    &mut recursion_depths,
-                    recursion_limit,
-                    inlining_map,
-                );
-            });
+            if is_parallel() {
```

Comment:
This block should be reverted as `par_for_each_in` should do this runtime check for us.

Reply:
Done